yt_dlp/extractor/upornia.py

```diff
@@ -0,0 +1,69 @@
+from .common import InfoExtractor
+from ..utils import traverse_obj
+import base64
+import json
+
+
+class UporniaIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?upornia\.com/videos/(?P<id>[0-9]+)/?.*/?.*'
+    _TESTS = [{
+        'url': 'https://upornia.com/videos/4451197/fit-girl-with-perfect-ass-in-black-yoga-pants-fucked-and-gets-creampie/',
+        'md5': 'd91eaa12f537d5092c357c174e5be5e4',
+        'info_dict': {
+            'id': '4451197',
+            'ext': 'mp4',
+            'title': 'Fit Girl With Perfect Ass In Black Yoga Pants Fucked And Gets Creampie',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': 'Anastasia Sorrentino',
+            'description': '',
+            # TODO more properties, either as:
+            # * A value
+            # * MD5 checksum; start the string with md5:
+            # * A regular expression; start the string with re:
+            # * Any Python type (for example int or float)
+        }
+    }]
+
+    def fixcyr(self, bla):
+        # https://stackoverflow.com/questions/14173421/use-string-translate-in-python-to-transliterate-cyrillic/14173535#14173535
+        # used SO base but extraced clean mapping
+        symbols = (u"МВАС,Е",
+                   u"MBAC/E")
+        tr = {ord(a): ord(b) for a, b in zip(*symbols)}
+        return bla.translate(tr)
```

Comment:
what exactly is this doing?

Reply:
It is replacing cyrillic characters that look identical to latin characters with their regular counterpart. It could probably be written cleaner like this, though im not sure where that `str.maketrans` problem comes into play.
```
    def deobfuscate(self, data):
        translation = str.maketrans('\u041c\u0412\u0410\u0421\u0415,', 'MBACE/')
        return data.translate(translation)
```